### PR TITLE
Add Decorator to my object

### PR DESCRIPTION
**Pull request by iambenkis**
## App name : School Library

In this initial step, I implement the classes to represent students and teachers.

### Checklist :1st_place_medal: 

- ```Person``` class create :heavy_check_mark: 
- ```Student``` class create :heavy_check_mark: 
- ```Teacher``` class create :heavy_check_mark: 
- ```Nameable``` class create (**Interface**) :heavy_check_mark: 
- Turn your ```Person``` class to ```Nameable``` :heavy_check_mark: 
-  ```Decorator``` base class prepared :heavy_check_mark: 
- ```CapitalizeDecorator``` and ```TrimmerDecorator``` prepared :heavy_check_mark: 

Happy reviewing :+1:

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,12 +1,61 @@
-class Person
+class Nameable
+  def initialize(name)
+    @nameable = name
+  end
```
Comment (mikethreels) [inline person.rb]: - [ ] no need to include the initialize method in this class because there are not instance variables used

File: person.rb [hunk 2]
```diff
@@ -1,12 +1,61 @@
...
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented methods '#{__method__}'"
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super
+  end
```
Comment (mikethreels) [inline person.rb]: - [ ] the attribute accessor should be created for the `@nameable` instance variable.

File: person.rb [hunk 3]
```diff
@@ -1,12 +1,61 @@
...
+
+  def correct_name
+    @nameable
+  end
+end
```
Comment (mikethreels) [inline person.rb]: - [ ] the `correct_name` method should return the result of the `correct_name` method of the `@namable`, which will look something like:
```
    def correct_name
      @nameable.correct_name
    end

```

File: person.rb [hunk 4]
```diff
@@ -1,12 +1,61 @@
...
+
+class TrimmerDecorator < Decorator
+  def initialize(nameable)
+    @nameable = nameable
+    super
+  end
+
+  def correct_name
+    return @nameable.correct_name.slice(0, 10) if @nameable.correct_name.length > 10
```
Comment (mikethreels) [inline person.rb]: - [ ] this method should return the first 10 characters of the `#nameable` variable however your code is returning the first 11.

File: person.rb [hunk 5]
```diff
@@ -1,12 +1,61 @@
...
+  def correct_name
+    @nameable
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def initialize(nameable)
+    @nameable = nameable
+    super
+  end
```
Comment (mikethreels) [inline person.rb]: - [ ] it is not necessary to use the initialize method in each class.

File: person.rb [hunk 6]
```diff
@@ -1,12 +1,61 @@
...
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def initialize(nameable)
+    @nameable = nameable
+    super
+  end
```
Comment (mikethreels) [inline person.rb]: - [ ] it is not necessary to use the initialize method in each class.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Benjamin, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codecaiine — APPROVED** [review]
Hi @iambenkis, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good project setup :heavy_check_mark: 
- No linters errors :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**